packages/wonder-blocks-badge/src/components/__tests__/badge-types.test.tsx

```diff
@@ -19,49 +19,57 @@ describe("Badge types", () => {
         it("should forward the ref to the root element", () => {
             // Arrange
             const ref = React.createRef<HTMLDivElement>();
-            render(<Component ref={ref} label="Badge label" />);
 
             // Act
-            const badge = screen.getByText("Badge label");
+            const {container} = render(
+                <Component ref={ref} label="Badge label" />,
+            );
 
             // Assert
-            expect(ref.current).toBe(badge);
+            // eslint-disable-next-line testing-library/no-node-access -- explicitly check the root element
```

Comment:
Because we wrap the label text in a `span` now, `screen.getByText` returns that `span` instead of the root element of the badge component. We now use `container.firstChild` to get the root element of the badge components

Reply:
Have you tried using `getByText(container, /badge label/i)` instead?

Reply:
I feel that it would be better to keep the rule enabled in that case, so we still signal that using this approach is a escape hatch. Silencing the rule would give the wrong signal in tests where we would be allowing using this approach more.

If `getByText` didn't work and testId is not possible (as Marcy suggests), I think it would be better to leave the tests as you created them in the first place.

Reply:
The rule continues to be enabled, but it's configured to allow specifically `container.firstChild` since that is the recommended approach for accessing the root element of a component. 

I'll take a look at switching the tests to use the test id!

Reply:
Thanks for the suggestions! I was able to use `getByTestId` instead 😄 For the unit tests specifically testing the `testId` prop, I continue to check the attribute on `container.firstChild` so we can confirm that functionality without using `getByTestId` (since we want to test it is on the right element)!